app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -242,7 +242,11 @@ function normalizeSignatureRequest(request: PPOMRequest): PPOMRequest {
     return request;
   }
 
-  const typedDataMessage = parseTypedDataMessage(request.params[1].toString());
+  const dataMessageParam = request.params[1];
+  const typedDataMessage =
```

Comment:
Can we do this inside `parseTypedDataMessage` to make it more flexible?

Plus we still want to perform the large value logic?

Reply:
Sure, moved to `parseTypedDataMessage`. Sorry Matthew, what do you mean by `still want to perform the large value logic?` ?